test/e2e/tests/grpc_test.go

```diff
@@ -30,3 +33,64 @@ func TestGRPC_Version(t *testing.T) {
 		require.Equal(t, version.BlockProtocol, res.Block)
 	})
 }
+
+func TestGRPC_GetBlockResults(t *testing.T) {
+	testNode(t, func(t *testing.T, node e2e.Node) {
+		if node.Mode != e2e.ModeFull && node.Mode != e2e.ModeValidator {
+			return
+		}
+
+		blocks := fetchBlockChain(t)
+
+		client, err := node.Client()
+		require.NoError(t, err)
+		status, err := client.Status(ctx)
+		require.NoError(t, err)
+
+		first := status.SyncInfo.EarliestBlockHeight
+		last := status.SyncInfo.LatestBlockHeight
+		if node.RetainBlocks > 0 {
+			first++
+		}
+
+		ctx, ctxCancel := context.WithTimeout(context.Background(), time.Minute)
+		defer ctxCancel()
+		gRPCClient, err := node.GRPCClient(ctx)
+		require.NoError(t, err)
+
+		for _, block := range blocks {
+			successCases := []struct {
+				expectedHeight int64
+				request        v1.GetBlockResultsRequest
+			}{
+				{first, v1.GetBlockResultsRequest{Height: first}},
+				{last, v1.GetBlockResultsRequest{}},
```

Comment:
In the endpoint definition it states that if no height is provided then the endpoint will fetch the latest block. The e2e test is indeed failing for the CI network setup (passes on simple) because of a race condition with the test setup. The test fails because the endpoint is fetching the latest endpoint which is greater than the value its being compared to (cached last block height).